amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java

```diff
@@ -50,6 +51,7 @@ public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProp
                 logger.debug(String.format("[consumer tag: %s] AMQP ack has been sent for message: %s", consumerTag, bodyStr));
             } catch (IOException e) {
                 logger.error(String.format("[consumer tag: %s] Failed to ack message %s", consumerTag, bodyStr));
+                throw new ChannelException(String.format("Failed to ack message '%s'", bodyStr), e);
```

Comment:
Taking action (logging) AND rethrowing an exception is considered to be anti-pattern: http://stackoverflow.com/questions/6639963/why-is-log-and-throw-condsidered-an-anti-pattern